acceptance/tests/facts/non_root_users_without_errors.rb

```diff
@@ -0,0 +1,34 @@
+test_name "C59196: running facter as a non-root user should not produce permission errors" do
+  tag 'risk:high'
+
+  # confine :except, :platform => 'aix' # bug FACT-1586
+
+  confine :except, :platform => 'windows' # this test is for unix systems
+  confine :except, :platform => 'aix' # has errors from the su command
+  confine :except, :platform => 'cisco' # has errors from the su command
+  confine :except, :platform => 'osx' # does not support managehome
+  confine :except, :platform => 'solaris' # does not work with managehome on solaris boxes
+  confine :except, :platform => 'eos-' # does not support user creation ARISTA-37
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  agents.each do |agent|
+    non_root_user = "nonroot"
+    facter_path = on(agent, "which facter").stdout.chomp
+
+    step "Agent #{agent}: create a #{non_root_user} user to run facter with" do
+      on(agent, "puppet resource user #{non_root_user} ensure=present managehome=true shell='#{user_shell(agent)}'")
```

Comment:
You would think that, but nope. The systems that fail to manage the home have a version of su(1) that complains that it can't access the parent directories of the working directory. I've changed it but you don't gain any coverage.